lightning/src/util/ser_macros.rs

```diff
@@ -1825,4 +1872,33 @@ mod tests {
 		assert_eq!(TuplesOnly::read(&mut none_data_read).unwrap(), None);
 		assert_eq!(none_data_read.position(), unknown_data_variant.len() as u64);
 	}
+
+	#[derive(Debug, PartialEq, Eq)]
+	struct ExpandedField {
+		// Old versions of LDK are presumed to have had something like:
+		// old_field: u8,
+		new_field: (u8, u8),
+	}
+	impl_writeable_tlv_based!(ExpandedField, {
+		(0, old_field, (legacy, u8, {
+			// Sadly the type-checker needs some help
+			let _: &(u8, u8) = &new_field;
+			if let Some(old_field) = old_field {
+				new_field.0 = old_field;
+			}
+		}, |us: &ExpandedField| Some(us.new_field.0))),
+		(1, new_field, required),
+	});
+
+	#[test]
+	fn test_legacy_conversion() {
+		let mut encoded = ExpandedField { new_field: (42, 43) }.encode();
+		assert_eq!(encoded, <Vec<u8>>::from_hex("0700012a01022a2b").unwrap());
+
+		// On read, the post-read action will run, using the old_field value to overwrite the
+		// new_field.
+		encoded[3] = 10;
+		let read = <ExpandedField as Readable>::read(&mut &encoded[..]).unwrap();
+		assert_eq!(read, ExpandedField { new_field: (10, 43) });
```

Comment:
Hi!
So, I noticed that changing the serialized value for `old_field` ends up updating `new_field` during deserialization, which makes sense for backward compatibility.

That said, I was wondering about this scenario: if `old_field` were public and `new_field` private, someone aware of `old_field` could modify its value, which would indirectly change the private `new_field`. Do we intend for a private field to be affected this way by something external?

This caught my eye while going through the code, and I’d love to better understand the reasoning behind this design choice. Do let me what you think! Thanks!